lib/pure/os.nim

```diff
@@ -2015,21 +2015,27 @@ proc tryMoveFSObject(source, dest: string): bool {.noWeirdTarget.} =
   ## Returns false in case of `EXDEV` error.
   ## In case of other errors `OSError` is raised.
   ## Returns true in case of success.
+  var isErr: bool
   when defined(windows):
     when useWinUnicode:
       let s = newWideCString(source)
       let d = newWideCString(dest)
-      if moveFileExW(s, d, MOVEFILE_COPY_ALLOWED or MOVEFILE_REPLACE_EXISTING) == 0'i32: raiseOSError(osLastError(), $(source, dest))
+      isErr = moveFileExW(s, d, MOVEFILE_COPY_ALLOWED or MOVEFILE_REPLACE_EXISTING) == 0'i32
     else:
-      if moveFileExA(source, dest, MOVEFILE_COPY_ALLOWED or MOVEFILE_REPLACE_EXISTING) == 0'i32: raiseOSError(osLastError(), $(source, dest))
+      isErr = moveFileExA(source, dest, MOVEFILE_COPY_ALLOWED or MOVEFILE_REPLACE_EXISTING) == 0'i32
   else:
-    if c_rename(source, dest) != 0'i32:
-      let err = osLastError()
-      if err == EXDEV.OSErrorCode:
-        return false
+    isErr = c_rename(source, dest) != 0'i32
+
+  if isErr:
+    let err = osLastError()
+    if err == EXDEV.OSErrorCode:
+      return false
+    else:
+      when defined(windows):
+        raiseOSError(err, $(source, dest))
       else:
-        # see whether `strerror(errno)` is redundant with what raiseOSError already shows
-        raiseOSError(err, $(source, dest, strerror(errno)))
+        raiseOSError(err, $(source ,dest, strerror(errno)))
```

Comment:
isn't `strerror(errno))` redundant with what raiseOSError already shows?

Reply:
@froehlichA click "resolve conversation" once you've resolved a comment (unless it's a push bash)